bot/exts/info/doc/_cog.py

```diff
@@ -0,0 +1,467 @@
+        """
+        Clear all internal symbol data.
+
+        All currently requested items are waited to be parsed before clearing.
+        """
+        for event in self._item_events.values():
+            await event.wait()
+        if self._parse_task is not None:
+            self._parse_task.cancel()
+            self._parse_task = None
+        self._queue.clear()
+        self._results.clear()
+        self._page_symbols.clear()
+        self._item_events.clear()
+
+
+class DocCog(commands.Cog):
+    """A set of commands for querying & displaying documentation."""
+
+    def __init__(self, bot: Bot):
+        self.base_urls = {}
+        self.bot = bot
+        self.doc_symbols: Dict[str, DocItem] = {}
+        self.item_fetcher = CachedParser()
+        self.renamed_symbols = set()
+
+        self.inventory_scheduler = Scheduler(self.__class__.__name__)
+        self.scheduled_inventories = set()
+
+        self.bot.loop.create_task(self.init_refresh_inventory())
+        self.bot.loop.create_task(doc_cache.delete_expired())
+
+    async def init_refresh_inventory(self) -> None:
+        """Refresh documentation inventory on cog initialization."""
+        await self.bot.wait_until_guild_available()
+        await self.refresh_inventory()
+
+    async def update_single(
+        self, api_package_name: str, base_url: str, inventory_url: str
+    ) -> bool:
+        """
+        Rebuild the inventory for a single package.
+
+        Where:
+            * `package_name` is the package name to use, appears in the log
+            * `base_url` is the root documentation URL for the specified package, used to build
+                absolute paths that link to specific symbols
+            * `inventory_url` is the absolute URL to the intersphinx inventory.
+
+        If the inventory file is currently unreachable,
+        the update is rescheduled to execute in 2 minutes on the first attempt, and 5 minutes on subsequent attempts.
+
+        Return True on success; False if fetching failed and was rescheduled.
+        """
+        self.base_urls[api_package_name] = base_url
+        package = await fetch_inventory(self.bot.http_session, inventory_url)
+
+        if not package:
+            delay = 2*60 if inventory_url not in self.scheduled_inventories else 5*60
+            log.info(f"Failed to fetch inventory, attempting again in {delay//60} minutes.")
+            self.inventory_scheduler.schedule_later(
+                delay,
+                api_package_name,
+                fetch_inventory(self.bot.http_session, inventory_url)
+            )
+            self.scheduled_inventories.add(api_package_name)
+            return False
+        with suppress(KeyError):
+            self.scheduled_inventories.discard(api_package_name)
+
+        for group, items in package.items():
+            for symbol, relative_doc_url in items:
+                if "/" in symbol:
+                    continue  # skip unreachable symbols with slashes
```

Comment:
After looking into it further I've removed the check. 
The current parsing can fetch most of them without failing, although in most cases it only spits out the hyperlink without any content, as usually the symbols only link to the page where the actual useful symbols exist; but there are things like `howto/deployment/wsgi/index:configuring the settings module` where it leads to actual content that's otherwise unreachable.
It adds back the couple tens of thousands symbols that'll probably never get looked up but that's an insignificant cost.